geode-redis/src/main/java/org/apache/geode/redis/internal/executor/connection/PingExecutor.java

```diff
@@ -31,9 +32,17 @@ public RedisResponse executeCommand(Command command,
       ExecutionHandlerContext context) {
     List<byte[]> commandElems = command.getProcessedCommand();
     byte[] result = PING_RESPONSE.getBytes();
+    byte[] subscribeResult = "".getBytes();
     if (commandElems.size() > 1) {
       result = commandElems.get(1);
+      subscribeResult = result;
     }
+
+    if (!context.getPubSub().findSubscribedChannels(context.getClient()).isEmpty()) {
```

Comment:
consider refactoring this so that we only compute the RedisResponse for subscribe if we find a client and then otherwise with (an else) compute the RedisResponse when not subscribed. I don't like all the logic we have initializing two different results and then we only end up using one or the other. I'd prefer that we only have state at the top that would be shared by either branch (for example commandElems). Once the RedisResponse is done we can then fall through to a command return.

Reply:
I refactored it. Check it out and let me know what you think.